101-vm-simple-windows-visualstudio2019/azuredeploy.json

```diff
@@ -0,0 +1,259 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "vmName": {
+            "type": "string",
+            "defaultValue": "[concat('vm', uniqueString(resourceGroup().name))]",
+            "metadata": {
+                "description": "The name of you Virtual Machine."
+            }
+        },
+        "VmSize": {
+            "type": "string",
+            "allowedValues": [
+                "Standard_D1_v2",
+                "Standard_D2_v2",
+                "Standard_D3_v2",
+                "Standard_D1_v2",
+                "Standard_D4_v2",
+                "Standard_D5_v2"
+        ],
+            "metadata": {
+                "description": "The size of the VM"
+            },
+        "defaultValue": "Standard_D2_v2"
+        },
+        "adminUsername": {
+            "type": "string",
+            "metadata": {
+                "description": "Username for the Virtual Machine."
+            }
+        },
+        "adminPassword": {
+            "type": "secureString",
+            "metadata": {
+                "description": "Password for the Virtual Machine. The password must be at least 12 characters long and have lower case, upper characters, digit and a special character (Regex match)"
+            }
+        },
+        "location": {
+            "type": "string",
+            "defaultValue": "[resourceGroup().location]",
+            "metadata": {
+                    "description": "Location for all resources."
+            }
+        },
+        "dnsLabelPrefix": {
+            "type": "string",
+            "defaultValue": "[toLower(concat('dns-',parameters('vmName')))]",
```

Comment:
use a globally unique default

Reply:
Now has this value:
"defaultValue": "[toLower(concat('dns-', uniqueString(parameters('vmName'))))]"

Reply:
use something descriptive of the vm for your prefix (dns is too generic)

Reply:
Sorry for not be so creative...

I came with this:

"defaultValue": "[toLower(concat('fqdn-', parameters('vmName')))]",

Is it better?